src/codelens/bazel_build_code_lens_provider.ts

```diff
@@ -126,50 +124,73 @@ export class BazelBuildCodeLensProvider implements vscode.CodeLensProvider {
   ): vscode.CodeLens[] {
     const result: vscode.CodeLens[] = [];
 
-    interface LensCommand {
-      commandString: string;
-      name: string;
-    }
-
-    const useTargetMap = queryResult.target
-      .map((t) => new QueryLocation(t.rule.location).line)
-      .reduce((countMap, line) => {
-        countMap.set(line, countMap.has(line));
-        return countMap;
-      }, new Map<number, boolean>());
-    // Sort targets by length first, then alphabetically
-    // This ensures shorter names (often main targets) appear first, with consistent ordering within each length group
+    // Sort targets alphabetically
     const sortedTargets = [...queryResult.target].sort((a, b) => {
-      const lengthDiff = a.rule.name.length - b.rule.name.length;
-      return lengthDiff !== 0
-        ? lengthDiff
-        : a.rule.name.localeCompare(b.rule.name);
+      return a.rule.name.localeCompare(b.rule.name);
     });
 
+    // Group targets by line number to handle multiple targets on same line
+    const targetsByLine = new Map<number, typeof sortedTargets>();
     for (const target of sortedTargets) {
       const location = new QueryLocation(target.rule.location);
+      const line = location.line;
+      if (!targetsByLine.has(line)) {
+        targetsByLine.set(line, []);
+      }
+      targetsByLine.get(line)?.push(target);
+    }
+
+    // Process each line's targets
+    for (const [, targets] of targetsByLine) {
+      // Use unified method for both single and multiple targets
+      this.createCodeLensesForTargets(targets, bazelWorkspaceInfo, result);
+    }
+
+    return result;
+  }
+
+  /**
+   * Creates CodeLens objects for targets on a line (handles both single and multiple targets)
+   */
+  private createCodeLensesForTargets(
+    targets: blaze_query.ITarget[],
+    bazelWorkspaceInfo: BazelWorkspaceInfo,
+    result: vscode.CodeLens[],
+  ): void {
+    const location = new QueryLocation(targets[0].rule.location);
+
+    // Group targets by action type using shared logic
+    const actionGroups = this.groupTargetsByAction(targets);
+
+    // Use unified method for all targets (handles both single and multiple)
+    this.createMultipleTargetCodeLenses(
```

Comment:
Minor: ...because I am having a hard time coming up with a good name: Maybe its better to avoid another helper function here and just directly paste the 4 statements directly in this function?